### PR TITLE
revert(TDI-38900) marketo api mode

### PR DESCRIPTION
**Pull request by undx**
**What is the current behavior?** (You can also link to an open issue here)

https://jira.talendforge.org/browse/TDI-38900

**What is the new behavior?**

Moved back apiMode to tMarketoConnection as discussed with @pteyssier.

**Please check if the PR fulfills these requirements**

- [X] The commit message follows Talend standard
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [X] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [X] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/423/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/423/)

File: components/components-marketo/src/main/java/org/talend/components/marketo/MarketoComponentProperties.java [hunk 1]
```diff
@@ -160,50 +142,19 @@ public Schema newSchema(Schema metadataSchema, String newSchemaName, List<Schema
...
-        return apiMode.getValue();
-    }
-
-    public void afterApiMode() {
-        getConnectionProperties().setApiMode(getApiMode());
+        return APIMode.REST.equals(getConnectionProperties().apiMode.getValue());
     }
 
     @Override
     public int getVersionNumber() {
```
Comment (jzhao-talend) [inline components/components-marketo/src/main/java/org/talend/components/marketo/MarketoComponentProperties.java]: I found that you revert the migration, So this method also need to be removed

File: components/components-marketo/src/main/java/org/talend/components/marketo/MarketoComponentProperties.java [hunk 2]
```diff
@@ -31,16 +28,14 @@
...
 import org.talend.components.common.SchemaProperties;
 import org.talend.components.marketo.tmarketoconnection.TMarketoConnectionProperties;
+import org.talend.components.marketo.tmarketoconnection.TMarketoConnectionProperties.APIMode;
 import org.talend.daikon.properties.presentation.Form;
 import org.talend.daikon.properties.property.Property;
-import org.talend.daikon.serialize.PostDeserializeSetup;
 import org.talend.daikon.serialize.migration.SerializeSetVersion;
 
 public abstract class MarketoComponentProperties extends FixedConnectorsComponentProperties
         implements MarketoProvideConnectionProperties, SerializeSetVersion {
```
Comment (jzhao-talend) [inline components/components-marketo/src/main/java/org/talend/components/marketo/MarketoComponentProperties.java]: I found that you revert the migration, So this implements also need to be removed

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/442/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/442/)

**Pull request merged**